src/google/cloud/ndb/_transaction.py

```diff
@@ -115,3 +115,136 @@ def _transaction_async(context, callback, read_only=False):
             raise
 
         return result
+
+
+def transactional(
+    retries=_retry._DEFAULT_RETRIES, read_only=False, xg=True, propagation=None
+):
+    """A decorator to run a function automatically in a transaction.
+
+    Usage example:
```

Comment:
I think this docstring may need some massaging to make pretty Sphinx docs, but as @cguardia is currently concerning himself with docs, I'll let him chime in here.